src/node/nodestate.h

```diff
@@ -1162,6 +1096,121 @@ namespace ccf
       secrets_view->put(version, secret_set);
     }
 
+    KeyShareInfo generate_key_share_info(
+      const std::vector<ccf::MemberPubInfo>& members_info)
+    {
+      auto share_wrapping_key_raw =
+        tls::create_entropy()->random(crypto::GCM_SIZE_KEY);
+      auto share_wrapping_key = crypto::KeyAesGcm(share_wrapping_key_raw);
+
+      // Encrypt initial ledger secrets with k_z
+      // Once sealing is completely removed, this can be called to the
+      // LedgerSecrets class directly
+      crypto::GcmCipher encrypted_ls(
+        network.ledger_secrets->get_secret(1)->master.size());
+
+      share_wrapping_key.encrypt(
+        encrypted_ls.hdr.get_iv(), // iv is always 0 here as the share wrapping
+                                   // key is never re-used for encryption
+        network.ledger_secrets->get_secret(1)->master,
+        nullb,
+        encrypted_ls.cipher.data(),
+        encrypted_ls.hdr.tag);
+
+      auto active_members = members_info.size();
+
+      // For now, the secret sharing threashold is set to the number of initial
+      // members
+      size_t k = active_members;
+
+      // For now, since members key shares are not yet encrypted, create
+      // share of dummy empty secret
+      SecretSharing::SecretToSplit secret_to_split = {};
+
+      auto shares = SecretSharing::split(secret_to_split, active_members, k);
+
+      assert(SecretSharing::combine(shares, k) == secret_to_split);
+
+      // For now, shares are recorded in plain text. Instead, they should be
+      // encrypted with each member's public encryption key
+      std::vector<std::vector<uint8_t>> encrypted_shares;
+      for (auto const& s : shares)
+      {
+        encrypted_shares.emplace_back(s.begin(), s.end());
+      }
+
+      return {encrypted_ls.serialise(), encrypted_shares};
+    }
+
+    std::vector<uint8_t> serialize_create_request(
```

Comment:
Just moved the following three functions down here to be in the private access specifier.